File: geometric-classes/lib/square.rb
class Square
  attr_reader :length

  def initialize (length, x = 0, y = 0)
    @length = length
    @x = x
    @y = y
  end

  def x
    @x
  end

  def y
    @y
  end

  def length
    @length
  end

  def width
    @length
  end

  def diameter
    @length * Math.sqrt(2)
  end

  def area
    @length ** 2
  end

  def perimeter
    @length * 4
  end

  def contains_point?(a, b)
    delta = length / 2.0
    if @x - delta <= a && a <= @x + delta && @y - delta <= b && b <= @y + delta
      return true
    else
      return false
    end
  end

end

File: geometric-classes-js/public/js/Square.js
class Square {
  constructor(length, x = 0, y = 0) {
    this.length = length;
    this.x = x;
    this.y = y;
  }
  area() {return(Math.pow(this.length));}
  perimeter() {return(4 * this.length);}
  contains_point(x,y) {
   return ( (x >= this.x-this.length/2) &&
   (x <= this.x+this.length/2) &&
   (y >= this.y - this.length/2) &&
   (y <= this.y+this.length/2) );
  }
}
